pimd/pim_ifchannel.c

```diff
@@ -771,13 +789,47 @@ void pim_ifchannel_join_add(struct interface *ifp, struct in_addr neigh_addr,
 
 	if (nonlocal_upstream(1 /* join */, ifp, upstream, sg, source_flags,
 			      holdtime)) {
+		if (PIM_DEBUG_TRACE) {
```

Comment:
You've added a bunch of good debug information, let's break these out into their own commit.  Also when breaking up please follow the commit instructions in http://docs.frrouting.org/projects/dev-guide/en/latest/workflow.html